third_party/amd/lib/TritonAMDGPUToLLVM/DotOpToLLVM/MFMA.cpp

```diff
@@ -430,6 +460,284 @@ struct DotOpMFMAConversionHelper {
   }
 };
 
+struct ScaledDotOpMFMAConversionHelper : DotOpMFMAConversionHelper {
```

Comment:
This helper has quite some duplication with the normal dot's. Can we try to extract some common base method/utilities? If possible, don't squash into this pull request; file a new pull request just does the NFC refactoring on the existing case to extract utilities out and rebase this pull request on top it to make life easier.

Reply:
Sure. I haven't taken closer look at that part since it worked fine. I thought it may introduce too many meaningless arguments to the utility function if extracting something out of a highly nested loop. Let me take a look.

Reply:
https://github.com/triton-lang/triton/pull/5845